integration/tests/cook/util.py

```diff
@@ -34,7 +35,18 @@ def has_docker_service():
     return os.path.exists('/var/run/docker.sock')
 
 
-class User(object):
+# Default user for multi-user test runs
+_default_user_name = 'root'
+_default_admin_name = 'root'
+_default_impersonator_name = 'poser'
+
+
+# Shell command used to obtain Kerberos credentials for a given test user
+_kerberos_missing_cmd =  'echo "MISSING COOK_KERBEROS_TEST_AUTH_CMD" && exit 1'
+_kerberos_auth_cmd = os.getenv('COOK_KERBEROS_TEST_AUTH_CMD', _kerberos_missing_cmd)
+
+
+class _AuthenticatedUser(object):
```

Comment:
Is it worth having this base class just for the logger lines? It seems like overkill to me.

Reply:
I was originally thinking there would be more common code in here... I ended up moving a lot of it into the UserFactory instead though. I kind of like having this—it defines the interface—but I guess it's not really needed in duck-typed Python... Do you feel really strongly that I should remove this?

Reply:
I don't know if I'd say really strongly, but I'd prefer to delete it. The duplicated logger code is more clear to me than calling out to the super methods in both children. There's no problem adding it back later if needed.